libs/cdk/src/lib/template-management/index.ts

```diff
@@ -0,0 +1,4 @@
+export * from './utils';
```

Comment:
do we really want a `utils` for each sub-folder? maybe we are able to combine all utils into a single utils folder/file. but this does not have to be in the scope of this PR

Reply:
The idea for utils per subfolder was to organize and share folder internal (not exported) functions across a folder.
Functions used in multiple folders would put in your suggested public utils. I refrained to to this so far because introducing it later leads to more thoughts on what we really need to export.

Reply:
To elaborate a bit more, here my idea on how we could start to organize dependencies:

if we organize CDK in sub-packages we have the following folder structure:

```terminal
cdk/src/lib/
├── sub-package1/
│   ├── index.ts
│   ├── model (single file or folder)
│   ├── utils (single file or folder)
│   └── feature (single file or folder)
├── sub-package2/
│   └── ...
└── core/
    ├── index.ts
    ├── model (single file or folder)
    └── utils (single file or folder)
```

- The `index.ts` in every folder defined the public exports and hides domain internal code away.
- The `core` folder contains abstract shared code.
- The `sub-package` folders define a specific domain e.g. `template-management` and contain domain dedicated code. 
- In some cases, we will reuse sub-packages in others. Here we should try to have as minimal relations as possible.

Reply:
The folder structure looks good to me so far. For code that is shared across `sub-package1, ... n` one might consider `common` instead of `core`? `core` suggests that it is required by all other subfolders, or might be required for the entire library to work. `common` sounds more like that code **may** but does not have to be used by other parts. What do you think?

Reply:
Very good! Yes.
Angular cdk e. G. Has a coerce folder. Which for sure is a common like thing.